### ISSUE TITLE
Weird characters showing up

### ISSUE THREAD

**Issue by andyperisho** [sharkdp/bat#98]
See pic below. I guess it is the lines that are not being drawn properly.
How can i fix it?

thx

![1526068458](https://user-images.githubusercontent.com/34944098/39944172-cd9a43bc-558f-11e8-8b16-a1b6929492a2.png)

**Comment by sharkdp** [comment]
Oh dear :smile: 

It looks like your terminal can not print the unicode characters `│`, `─`, `┬`, etc. - yes. Which terminal (and which font?) do you use?

I wonder if there is some way to detect this. `bat` could then maybe fall back to ASCII characters in this case.

**Comment by andyperisho** [comment]
i'm using termite.
I think the font I'm using is Hack. I'm on arch linux. how do I check that?

**Comment by andyperisho** [comment]
Nevermind, it had to do with my locale which was not set. Thx for the help!

**Comment by sharkdp** [comment]
Thank you for the feedback.

**Comment by Berjou** [comment]
Hi, I got the same problem with urxvt. However, my locale is set.

```
$ locale
LANG=en_US.utf8
LC_CTYPE="en_US.utf8"
LC_NUMERIC="en_US.utf8"
LC_TIME="en_US.utf8"
LC_COLLATE="en_US.utf8"
LC_MONETARY="en_US.utf8"
LC_MESSAGES="en_US.utf8"
LC_PAPER="en_US.utf8"
LC_NAME="en_US.utf8"
LC_ADDRESS="en_US.utf8"
LC_TELEPHONE="en_US.utf8"
LC_MEASUREMENT="en_US.utf8"
LC_IDENTIFICATION="en_US.utf8"
LC_ALL=
```

```
$ bat --version  
bat 0.3.0
```

Any idea on how to fix this?
Thx.

**Comment by sharkdp** [comment]
Does this work for you?
``` bash
printf "\xE2\x94\x82\n"
```

It shoud print [this unicode character](https://unicode-table.com/en/2502/): `│`.

**Comment by Berjou** [comment]
Yes it does.

**Comment by sharkdp** [comment]
Could you please show a screenshot of
```
echo "\xE2\x94\x82" | bat
```


I cannot reproduce this with `urxvt` (although I did not configure the font).

**Comment by Berjou** [comment]
There is the result with urxvt:
![2018-05-19-063534_1363x667_scrot](https://user-images.githubusercontent.com/10575885/40264977-4603e138-5b2f-11e8-9d4b-9ede8205ffb9.png)

And there is the same result with a default xterm:
![2018-05-19-063624_1361x514_scrot](https://user-images.githubusercontent.com/10575885/40264979-573c0bba-5b2f-11e8-9191-c778b874b46d.png)

**Comment by ghuls** [comment]
I noticed the same issue with bat 0.3.0 from crates.io.

The following works fine:
```
bat file.txt
```

This doesn't (launch bat with pager manually with same arguments as in the git version:
```
bat file.txt | less --RAW-CONTROL-CHARS --no-init --quit-if-one-screen
```

This works:
```
bat file.txt | less --raw-control-chars --no-init --quit-if-one-screen
```

Less manpage:
```
       -r or --raw-control-chars
              Causes "raw" control characters to be displayed.  The default is to display control characters using the caret notation; for example, a control-A (octal 001) is displayed as "^A".   Warn‐
              ing:  when  the  -r option is used, less cannot keep track of the actual appearance of the screen (since this depends on how the screen responds to each type of control character).  Thus,
              various display problems may result, such as long lines being split in the wrong place.

       -R or --RAW-CONTROL-CHARS
              Like -r, but only ANSI "color" escape sequences are output in "raw" form.  Unlike -r, the screen appearance is maintained correctly in most  cases.   ANSI  "color"  escape  sequences  are
              sequences of the form:

                   ESC [ ... m

              where  the "..." is zero or more color specification characters For the purpose of keeping track of screen appearance, ANSI color escape sequences are assumed to not move the cursor.  You
              can make less think that characters other than "m" can end ANSI color escape sequences by setting the environment variable LESSANSIENDCHARS to the list of characters which can end a color
              escape sequence.  And you can make less think that characters other than the standard ones may appear between the ESC and the m by setting the environment variable LESSANSIMIDCHARS to the
              list of characters which can appear.

```
Pull request: https://github.com/sharkdp/bat/pull/143

**Comment by ghuls** [comment]
For me, `less --RAW-CONTROL-CHARS` was not working on this version of less:
```
$ less -V
less 436
Copyright (C) 1984-2009 Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Homepage: http://www.greenwoodsoftware.com/less
```
On another server, we have another version of less, and there  `less --RAW-CONTROL-CHARS` works:
```
$ less -V
less 458 (GNU regular expressions)
Copyright (C) 1984-2012 Mark Nudelman

less comes with NO WARRANTY, to the extent permitted by law.
For information about the terms of redistribution,
see the file named README in the less distribution.
Homepage: http://www.greenwoodsoftware.com/less
```

**Comment by ghuls** [comment]
@sharkdp You are right about the #143 pull request. I force pushed the probably proper fix to that pull request.

**Comment by sharkdp** [comment]
> @sharkdp You are right about the #143 pull request. I force pushed the probably proper fix to that pull request now, so if you can check it again.

@ghuls Thanks! You seem to have a different issue than @Berjou though, right (since `bat file.txt` works for you)?

**Comment by ghuls** [comment]
@sharkdp `bat file.txt` worked with the version of bat on crates.io (which did not use a pager yet) not with the git version.

**Comment by Berjou** [comment]
Setting the `LESSCHARSET` solved the issue for me. Thank you for your time.

**Comment by sharkdp** [comment]
@ghuls I see. Thank you for the clarification and for #143!

@Berjou Thank you for checking!

**Comment by sharkdp** [comment]
Fixed in v0.4
